Build failure: libetonyek (cross-compilation)

- **Package:** [pkgs/by-name/li/libetonyek/package.nix](https://github.com/NixOS/nixpkgs/blob/nixpkgs-unstable/pkgs/by-name/li/libetonyek/package.nix)
- **Problematic file and line:** [src/lib/IWORKToken.cpp#L48-L49](https://github.com/LibreOffice/libetonyek/blob/06467aeb7b4a7082d524d3433c166e12278a3104/src/lib/IWORKToken.cpp#L48-L49)
- **Upstream issue about this failure:** https://bugs.documentfoundation.org/show_bug.cgi?id=167179

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

```
nix-build '<nixpkgs>' -I nixpkgs=channel:nixos-unstable --arg crossSystem '{ config = "aarch64-unknown-linux-gnu"; }' -A libetonyek
```

### Relevant log output

```console
IWORKToken.cpp: In member function 'virtual int libetonyek::{anonymous}::Tokenizer::queryId(const char*) const':
IWORKToken.cpp:48:9: error: 'Token' does not name a type
   48 |   const Token *const token = Perfect_Hash::in_word_set(name, (unsigned) length);
      |         ^~~~~
IWORKToken.cpp:49:10: error: 'token' was not declared in this scope
   49 |   return token ? token->id : 0;
      |          ^~~~~
IWORKToken.cpp:45:16: warning: unused variable 'length' [-Wunused-variable]
   45 |   const size_t length = strlen(name);
      |                ^~~~~~
make[4]: *** [Makefile:1729: libetonyek_internal_la-IWORKToken.lo] Error 1
```

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.15.1, NixOS, 25.05 (Warbler), 25.05.803579.70c74b02eac4`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.3`
 - channels(root): `"nixos-25.05, nixos-hardware"`
 - nixpkgs: `/nix/store/gdz4yp5sjq3b1q0g9ad9hicqsx1alvif-nixos-25.05/nixos`

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/li/libetonyek/package.nix
+++ b/pkgs/by-name/li/libetonyek/package.nix
@@ -29,13 +29,13 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [
     autoreconfHook
     pkg-config
+    gperf
   ];
 
   buildInputs = [
     boost
     cppunit
     glm
-    gperf
     liblangtag
     librevenge
     libxml2

